zerver/lib/actions.py

```diff
@@ -931,6 +931,26 @@ def do_send_messages(messages_maybe_none):
             event['sender_queue_id'] = message['sender_queue_id']
         send_event(event, users)
 
+        # Trigger hotspot when user replies to welcome bot for the first time.
+        welcome_bot_user_profile = get_system_bot(settings.WELCOME_BOT)
+        is_personal_message = message['message'].recipient.type == Recipient.PERSONAL
+        recipient_user_profile = None if not is_personal_message else get_user_profile_by_email(
+            get_display_recipient(message['message'].recipient)[0]['email'])
+        is_message_to_welcome_bot = all([not message['message'].sender.is_bot,
+                                        is_personal_message,
+                                        welcome_bot_user_profile == recipient_user_profile])
+        if is_message_to_welcome_bot:
+            sender_hotspot_field = UserHotspot.objects.get(
+                user=message['message'].sender,
+                hotspot='welcome_bot_first_reply')
+            if not sender_hotspot_field.has_welcome_bot_replied_to_user:
+                internal_send_private_message(message['realm'],
+                                              welcome_bot_user_profile,
+                                              message['message'].sender,
+                                              ":tada: I'm out for now, but enjoy the rest of your Zulip experience!")
+                sender_hotspot_field.has_welcome_bot_replied_to_user = True
+                sender_hotspot_field.save()
```

Comment:
@showell can help with this, but we want to avoid doing an extra database query in this codepath for every message we send.  There's a similar issue with the feedback bot that we're discussing how to resolve.  

Structurally, I think we want most of the logic here to go into a new function `handle_welcome_bot_reply` or something similar, maybe in `zerver/lib/onboarding.py` or a similar place.  Otherwise, we end up cluttering this code path a lot.

I would also move this if statement next to the feedback-bot one, since while the use of queuing is different, it's otherwise very analogous.